MergerLogic/Batching/Tile.cs

```diff
@@ -25,26 +26,26 @@ public int Y
             internal set;
         }
 
-        public TileFormat? Format { get; internal set; }
+        public TileFormat Format { get; internal set; }
 
         private byte[] _data;
 
-        public Tile(int z, int x, int y, byte[] data, TileFormat? format = null)
+        public Tile(int z, int x, int y, byte[] data)
         {
             this.Z = z;
             this.X = x;
             this.Y = y;
             this._data = data;
-            this.Format = format;
+            this.Format = ImageFormatter.GetTileFormat(data) ?? throw new ValidationException($"Cannot create tile {this}, data is in invalid format");
```

Comment:
please put it on "this._data = data;" - as ImageFormatter.GetTileFormat will throw on null reference

Reply:
The issue is when trying to extract the format, not when assigning data.
You shouldn't get `null` data.